.circleci/parse_yaml.py

```
@@ -0,0 +1,273 @@
+            self.data_type = self.yaml_data['data_info']['data_type']
+        else:
+            logging.error(f' there is no data_type indicated.')
+            sys.exit(1)
+
+
+    def download(self):
+        """Check if provided weblink to the file exists. 
+        Download if verified.
+        """
+        path = self.yaml_data['data_file']['location']['url']
+        self.downloaded_file = path.split('/')[-1]
+
+        outdir_root = 'submitted_data/'
+        pathlib.Path(outdir_root).mkdir(exist_ok=True)
+        self.outdir = outdir_root + str(self.data_type) + '/'
+        pathlib.Path(self.outdir).mkdir(exist_ok=True)
+
+        if verify_weblink(path):
+            logging.info(f' starting download')
+            download_file(path, self.outdir)
+            self.hash = self.yaml_data['data_file']['location']['md5']
+        else:
+            logging.error(f' file unavailable')
+            sys.exit(1)
+
+
+    def bulk_download(self):
+        """Bulk download option assumes one of the following file types:
+        ['tgz', 'tar.gz', 'tar.bz2', 'tar.xz']
+        """
+        import os
+        import tarfile
+
+        self.download()
+
+        if tarfile.is_tarfile(self.downloaded_file):
+            tarred_data = tarfile.open(
+                self.downloaded_file, 
+                mode='r:*'
+            )
+            for member in tarred_data.getmembers():
+                if member.isreg():
+                    member.name = os.path.basename(member.name)
+                    tarred_data.extract(member, path=self.outdir)
+    
+
+    def check_hash(self):
+        """Compare md5 of the downloaded file to the provided value"""
+        if validate_hash(self.downloaded_file, self.hash):
+            logging.info(f' data is ready for cimr processing.')
+            return True
+        else:
+            logging.error(f' provided md5 hash didn\'t match the downloaded file.')
+            return False
```

When the hash doesn't match, do you think we should raise an exception immediately to terminate python and CircleCI process?

---

yes